src/sdam/message_manager.rs

```diff
@@ -1,51 +1,47 @@
-use std::{sync::Arc, time::Duration};
+use std::time::Duration;
 
-use tokio::sync::watch::{self, Receiver, Sender};
+use tokio::sync::broadcast::{self, Receiver, Sender};
```

Comment:
While I was debugging some issues with this due to some quirks with the `watch` channel, Patrick suggested that the `broadcast` channel might make more sense here, since we fundamentally want a multi-producer, multi-consumer channel. Switching to that makes the code much cleaner, and since we didn't use the internals of this type from outside the module, it also didn't require changing any outside code.